client-ui/src/app/components/dashboard/dashboard.component.ts

```diff
@@ -13,200 +13,132 @@
-          this.applyFilters()
-        }
-      )
-    )
-
-    this.subscriptions.push(
-      fromEvent<any>(this.searchInput.nativeElement, 'keyup')
-        .pipe(
-          tap(_ => this.loading = true),
-          map(event => event.target.value),
-          debounceTime(400)
-        ).subscribe(
-          searchTerm => {
-            this.queryParams = { ...this.queryParams, offset: 0, searchTerm: searchTerm }
-            this.applyFilters()
-          }
-        )
-    )
-
-  }
-
-  private applyFilters(): void {
-    if (this.useDateRange) {
-      this.store.dispatch(new RouterAction.ReplaceUrlState(this.queryParams))
-    }
-    this.store.dispatch(new ExecutionEventsActions.Get({ ...this.queryParams, asAtTime: this.asAtTime }))
-    this.loading = false
+  private loadExecutionEvents(dashboardState): void {
+    const dateRange = dashboardState.filters.dateRange
+    this.store.dispatch(new ExecutionEventsActions.Get(
+      {
+        asAtTime: this.asAtTime,
+        timestampStart: dateRange && dateRange[0],
+        timestampEnd: dateRange && dateRange[1],
+        offset: dashboardState.pagination.page,
+        sortName: dashboardState.sort.field,
+        sortDirection: dashboardState.sort.direction,
+        searchTerm: dashboardState.filters.searchQuery,
+      }))
   }
 
-  public onUseDateRangeChange(value): void {
-    this.useDateRange = value
-    this.queryParams = !this.useDateRange && {}
-    this.store.dispatch(new RouterAction.ReplaceUrlState(this.queryParams))
+  ngOnDestroy(): void {
+    this.subscriptions.forEach(s => s.unsubscribe())
   }
 
   public onSelect(event): void {
     const executionEventId = event.selected[0].executionEventId
     const params = {} as RouterStateUrl
-    params.queryParams = { "executionEventId": executionEventId }
+    params.queryParams = {"executionEventId": executionEventId}
     params.url = "/app/lineage-overview/"
     this.store.dispatch(new RouterAction.Go(params))
   }
 
+  public onFilterByDateSwitchToggle(value): void {
+    const dateRange = value
+      ? this.executionEvents.totalDateRange as DateRange
+      : undefined;
+    this.store.dispatch(new DashboardActions.SetDateRange(dateRange))
+  }
 
-  public getFrameworkImg(frameworkName: String): String {
+  public onDateRangeSelected(range): void {
+    this.store.dispatch(new DashboardActions.SetDateRange(range))
+  }
+
+  public onPageChange(page): void {
+    this.store.dispatch(new DashboardActions.SetPageNumber(page.offset))
+  }
+
+  public onSortChange(sort: TableSort): void {
+    this.store.dispatch(new DashboardActions.SetSortOrder(sort.prop, sort.dir))
+  }
+
+  public readonly onSearchTermChange: (_: KeyboardEvent) => void = _.debounce(
+    (input: KeyboardEvent) =>
+      this.store.dispatch(new DashboardActions.SetSearchQuery((input.target as HTMLInputElement).value)),
+    SEARCH_TERM_UPDATE_DELAY)
+
+  getFrameworkImg(frameworkName: String): String {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The class method 'getFrameworkImg' must be marked either 'private', 'public', or 'protected'](https://app.codacy.com/app/ABSA_OSS/spline/pullRequest?prid=4491035)